landoapi/projects.py

```diff
@@ -53,15 +53,22 @@ def project_search(
     if not project_phids:
         return {}
 
+    project_phids = list(project_phids)
+    project_phids.sort()
+    cache_key = ",".join(project_phids)
+
+    if cache.has(cache_key):
+        return cache.get(cache_key)
+
     projects = phabricator.call_conduit(
         "project.search", constraints={"phids": project_phids}
     )
-    return result_list_to_phid_dict(phabricator.expect(projects, "data"))
+    result = result_list_to_phid_dict(phabricator.expect(projects, "data"))
+    cache.set(cache_key, result, timeout=DEFAULT_CACHE_KEY_TIMEOUT_SECONDS)
+    return result
 
 
-def get_project_phid(
-    project_slug: str, phabricator: PhabricatorClient, allow_empty_result: bool = True
-) -> Optional[str]:
+def get_project_phid(project_slug, phabricator, allow_empty_result=True):
```

Comment:
Type hints have been removed.

Reply:
Will re-add those, may have been a result of hunk-patching from the revision worker patch and I missed non-functional changes.